src/test/java/com/arpnetworking/test/TelemetryClient.java

```diff
@@ -0,0 +1,276 @@
+            final Consumer<Double> callback) {
+        try {
+            _webSocket.sendTextFrame(
+                    OBJECT_MAPPER.writeValueAsString(
+                            ImmutableMap.of(
+                                    "command", "subscribeMetric",
+                                    "service", service,
+                                    "metric", metricName,
+                                    "statistic", statistic.getName())));
+            _callbacks.put(String.format("%s/%s/%s", service, metricName, statistic.getName()), callback);
+        } catch (final JsonProcessingException e) {
+            LOGGER.warn()
+                    .setMessage("Failed to subscribe to metric")
+                    .setThrowable(e)
+                    .log();
+        }
+    }
+
+    /**
+     * Stop receiving data for the specified time series.
+     *
+     * @param service the service name
+     * @param metricName the metric name
+     * @param statistic the statistic
+     */
+    public void unsubscribe(
+            final String service,
+            final String metricName,
+            final Statistic statistic) {
+        try {
+            _callbacks.remove(String.format("%s/%s/%s", service, metricName, statistic.getName()));
+            _webSocket.sendTextFrame(
+                    OBJECT_MAPPER.writeValueAsString(
+                            ImmutableMap.of(
+                                    "command", "unsubscribeMetric",
+                                    "service", service,
+                                    "metric", metricName,
+                                    "statistic", statistic.getName())));
+        } catch (final JsonProcessingException e) {
+            LOGGER.warn()
+                    .setMessage("Failed to unsubscribe to metric")
+                    .setThrowable(e)
+                    .log();
+        }
+    }
+
+    private static String getEnvOrDefault(final String name, final String defaultValue) {
+        return getEnvOrDefault(name, defaultValue, Function.identity());
+    }
+
+    private static <T> T getEnvOrDefault(final String name, final T defaultValue, final Function<String, T> map) {
+        @Nullable final String value = System.getenv(name);
+        return value == null ? defaultValue : map.apply(value);
+    }
+
+    private TelemetryClient(final String host, final int port) throws ExecutionException, InterruptedException {
+        final AsyncHttpClient asyncHttpClient = Dsl.asyncHttpClient();
+        _webSocket = asyncHttpClient.prepareGet(String.format("ws://%s:%d/telemetry/v2/stream", host, port))
+                .execute(
+                        new WebSocketUpgradeHandler.Builder()
+                                .addWebSocketListener(new TelemetryWebSocketListener(host, port, _callbacks))
+                                .build())
+                .get();
+
+        _keepAliveExecutor = Executors.newScheduledThreadPool(1);
+        _keepAliveExecutor.schedule(
```

Comment:
Does this action want to be recurring? It looks to me like it will only run once.

Reply:
Good catch! Fixed.